src/azure/Model/CompositeTypeRba.cs

```diff
@@ -68,22 +68,21 @@ public override string GetBaseTypeName()
         }
 
         /// <summary>
-        /// Checks if the provided definition of models 'Resource'/'SubResource' matches the standard definition,
-        /// as defined in MsRestAzure. For other models, it returns false.
+        /// Checks if the provided definition of model matches the standard definition of 'Resource'/'SubResource',
+        /// as defined in MsRestAzure.
         /// </summary>
-        /// <param name="model">to be validated</param>
-        /// <returns></returns>
+        /// <param name="model">CompositeType model to be validated.</param>
+        /// <returns><c>true</c> if model matches standard name and definition of Resource or SubResource, <c>false</c> otherwise.</returns>
         public static bool IsResourceModelMatchingStandardDefinition(CompositeType model)
         {
             string modelName = model.Name.ToString();
-            if (modelName.EqualsIgnoreCase("SubResource") &&
-                model.Properties.All(property => subResourceRegEx.IsMatch(property.Name.ToString())))
+            if (!resourceOrSubResourceRegEx.IsMatch(modelName))
             {
-                return true;
+                return false;
             }
 
-            if(modelName.EqualsIgnoreCase("Resource") &&
-               model.Properties.All(property => resourceRegEx.IsMatch(property.Name.ToString())))
+            if (model.Properties.All(property => subResourceRegEx.IsMatch(property.Name.ToString())) || 

```

Comment:
are name checks enough for this? I believe "read-only" properties may vary, not sure why, but wonder if it can cause problems... for example: https://github.com/Azure/azure-rest-api-specs/blob/97e69db426af23a0cdfde5e5df748565ff1070b8/specification/network/resource-manager/Microsoft.Network/2017-09-01/network.json#L99 and https://github.com/Azure/azure-rest-api-specs/blob/bc175dbf36fd6d8e4ccac18a3bfd102d43fc6672/specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json#L6828